### PR TITLE
php-fpm

### PR DESCRIPTION
**Pull request by nrwiersma**
Hi,

I have ported php-fpm from your python module.

Please let me know if there is anything i missed in the process.

### CODE REVIEW

**Comment by ilyam8** [comment]
@nrwiersma thx! will review the PR tomorrow 😉

File: modules/phpfpm/client.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		resp.Body.Close()
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%s returned HTTP status %d", req.URL, resp.StatusCode)
+	}
+
+	s := &status{}
+	if err := c.dec(resp.Body, s); err != nil {
+		return nil, fmt.Errorf("error parsing response")
```
Comment (ilyam8) [inline modules/phpfpm/client.go]: > fmt.Errorf("error parsing response")

forgot to add the error

File: modules/phpfpm/phpfpm.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	}
+
+	data := stm.ToMap(status)
+	if len(status.Processes) == 0 {
+		return data
+	}
+
+	statProcesses(data, status.Processes, "ReqDur", func(p proc) int64 { return p.Duration })
+	statProcesses(data, status.Processes, "ReqCpu", func(p proc) int64 { return int64(p.CPU) })
+	statProcesses(data, status.Processes, "ReqMem", func(p proc) int64 { return p.Memory })
```
Comment (ilyam8) [inline modules/phpfpm/phpfpm.go]: please move collect logic to the `collect.go`

https://github.com/netdata/go.d.plugin/blob/e2059a14dc50d2c4acd86e64dfde4b5f1f99a15c/modules/vsphere/vsphere.go#L187-L199

File: modules/phpfpm/decode.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	"strings"
+)
+
+type decoder func(r io.Reader, s *status) error
+
+func decodeJSON(r io.Reader, s *status) error {
+	return json.NewDecoder(r).Decode(s)
+}
+
+func decodeText(r io.Reader, s *status) error {
```
Comment (ilyam8) [inline modules/phpfpm/decode.go]: overall decodeText func is hard to follow, how about to split it into several smaller ones?

File: modules/phpfpm/phpfpm.go (hunk 2 above)
Comment (nrwiersma) [reply modules/phpfpm/phpfpm.go]: Is there any particular reason for this? From a Go perspective, this is not needed or probably wanted. You are merely splitting code a) into a function and b) into a file when the logic is in fact really simple.

Comment (ilyam8) [reply modules/phpfpm/phpfpm.go]: The reason is having `collect.go` as an entry point for collecting logic in all modules
